### ISSUE TITLE
Linux hist triggers support

### ISSUE THREAD

**Issue by brendangregg** [intelsdi-x/snap#958]
This should be in Linux 4.7-rc1 (not released yet), contributed by Tom Zanussi (Intel). It's a new tracing/data aggregation capability that is built into the kernel, and used via the ftrace (now called "trace") framework.

Search for "hist:" in https://github.com/torvalds/linux/blob/master/Documentation/trace/events.txt

It's similar to BPF, although more specific use cases. I'd suspect hist triggers it could also be used to implement some of the same things as BPF can, eg, (from https://github.com/intelsdi-x/snap/issues/800):

- Disk I/O latency histograms per-second (which could then be visualized as a latency heat map)
- File system latency histograms
- Run queue latency histograms
- Sampled stack traces (for generation into flame graphs)
- etc.

**Comment by lynxbat** [comment]
This is great. We need to use #800 and this along with our plans to make a more holistic plugin for the Linux OS telemetry.

Would you mind reviewing an RFC on a plugin including these features when it is ready @brendangregg?

**Comment by brendangregg** [comment]
Yes, I can review. I posted a few more hist triggers examples here: http://www.brendangregg.com/blog/2016-06-08/linux-hist-triggers.html
